### PR TITLE
OSPP 2021-结项报告-基于 Itchat 为 Python Wechaty 实现社区首个 Python 语言的本地 Puppet

### PR DESCRIPTION
**Pull request by ly1ex**
Please make sure to check the following boxes creating your pull request, thank you very much!

- [x] Add Blog Header, including title, author, image, categories, tags, etc.  

    For example:  

    ```md
    ---
    title: Your Great Blog Title
    author: your_github_id
    categories: project
    tags:
      - python
      - padlocal
    image: /assets/2021/02-your-blog-slug/teaser.jpg
    ---

    One line bio

    Your blog post content Your blog post content
    Your blog post content Your blog post content
    Your blog post content Your blog post content
    ```

    > It will be easier to start drafting your post by using a existing post file as your template, for example, [this one](https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_posts/2021-01-14-wechaty-puppet-service.md)

- [x] Keep all filenames lowercase, and use `-` to connect words instead of space. e.g. `2017-10-06-wechat-pc-impactor.md` instead of `2017-10-06-WeChat PC Impactor.md`
- [x] Embed video/pdf from YouTube/Bilibili/local (if there's any) by using `{% include iframe.html src="URL_or_PATH" %}`. (Learn more about how to embed file/url in the post by reading this [blog post](https://wechaty.js.org/2020/08/24/add-video-to-wechaty-blog/)
- [x] Save photo to local before publishing, to a folder named by `${MONTH}-${YOUR_BLOG_SLUG}/` under `/assets/2021/` directory.
- [x] Select a beautiful and meaningful teaser image for your blog
- [x] Create your contributor profile (if you are a first time contributor)

    For excample:

    > jekyll/_contributors/lijiarui.md

    ```yaml
    ---
    name: Jiarui LI (李佳芮)
    site: https://lijiarui.github.io
    bio: 句子互动创始人 & CEO，微软人工智能最具价值专家 (AI MVP)
    avatar: /jekyll/assets/contributors/lijiarui/avatar.png
    email: rui@juzi.bot
    twitter: https://weibo.com/u/2175505900
    ---
    ```

  Here is the example file:

  - lijiarui.md: <https://github.com/wechaty/wechaty.js.org/blob/master/jekyll/_contributors/lijiarui.md>
  - It will shown like this on wechaty website: <https://wechaty.js.org/contributors/lijiarui/>

To learn more about the contribute guideline, visit: <https://wechaty.js.org/docs/contributor-program/>

### CODE REVIEW

**Comment by ly1ex** [comment]
明日上传视频后更新视频链接～

**Review by wj-Mcat — APPROVED** [review]
LGTM

**Comment by ly1ex** [comment]
项目地址：wechaty/summer#75

File: jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    - L3阶段开发 10天
+    - L4阶段开发 15天
+    - L5阶段开发 15天
+    - 结项验收
+
+## 项目总结
+
+- 项目产出：
+  - 产出1：Python Wechaty Puppet Itchat：[https://github.com/wechaty/python-wechaty-puppet-itchat](https://github.com/wechaty/python-wechaty-puppet-itchat)，代码已合并到社区
+  - 产出2：PyPi 包：[https://pypi.org/project/wechaty-puppet-itchat/](https://pypi.org/project/wechaty-puppet-itchat/)，已可以正常使用
```
Comment (lijiarui) [inline jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md:69]: Please add your mentor (@wj-Mcat and @huan  ) as Maintainers to https://pypi.org/project/wechaty-puppet-itchat

**Review by lijiarui — CHANGES_REQUESTED** [review]
Good job! Looking forward to  [wechaty-puppet-cq](https://github.com/lyleshaw/wechaty-puppet-cq)! Cheers!

Please:
1. Change as I request
2. Add Chinese version video link as soon as possible.

File: jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md (hunk 1 above)
Comment (ly1ex) [reply jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md:69]: > Please add your mentor (@wj-Mcat and @huan ) as Maintainers to https://pypi.org/project/wechaty-puppet-itchat

Done~

Comment (lijiarui) [reply jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md:69]: Good job! When will you add the Chinese version video link? 

Only when you add a Chinese version link, then I would merge this PR.

Comment (ly1ex) [reply jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md:69]: > Good job! When will you add the Chinese version video link?
> 
> Only when you add a Chinese version link, then I would merge this PR.

I'll try my best to complete the video release tonight (due to my PyCharm license expired and internet speed problems, there may be a delay)

**Comment by lijiarui** [comment]
It seems your PR failed the test, please make GitHub action green first.

**Comment by ly1ex** [comment]
> It seems your PR failed the test, please make GitHub action green first.

I'll update PDF file to pass CI soon

**Comment by ly1ex** [comment]
> It seems your PR failed the test, please make GitHub action green first.

@lijiarui 

I have some questions about `assets-pdf-size.spec.ts`.

It requires `PDF size should be fit for the web (no more than 3MB)`, but at the same time it says `the maximum limitation: size<=1048576`.

I have tried my best to compress my PDF to 2.4MB, it seems that I have no way to compress it to 1MB.

**Comment by lijiarui** [comment]
Hey @lyleshaw 

I guess there is a typo here about 3MB, it should be less than 1MB. We will appreciate if you can create a PR for us:

```
It requires PDF size should be fit for the web (no more than 3MB), but at the same time, it says the maximum limitation: size<=1048576.
```

You can try one of these suggestions:

1. Use https://www.ilovepdf.com/compress_pdf to compress to less than 1MB.
2. Upload to [Google Slides](https://docs.google.com/presentation/u/0/) and use Google Slides to embed your pdf here.

**Comment by ly1ex** [comment]
> Hey @lyleshaw
> 
> I guess there is a typo here about 3MB, it should be less than 1MB. We will appreciate if you can create a PR for us:
> 
> ```
> It requires PDF size should be fit for the web (no more than 3MB), but at the same time, it says the maximum limitation: size<=1048576.
> ```
> 
> You can try one of these suggestions:
> 
> 1. Use https://www.ilovepdf.com/compress_pdf to compress to less than 1MB.
> 2. Upload to [Google Slides](https://docs.google.com/presentation/u/0/) and use Google Slides to embed your pdf here.

Done

File: jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  - Python Wechaty Puppet Itchat
+    - 在结项后，我仍计划持续维护这一 Puppet，并计划将 Itchat 的代码进行重构
+  - Python Wechaty
+    - 我计划之后也持续参与到 Python Wechaty 生态的建设中去 （包括但不限于：为 Python Wechaty 贡献代码、为 python-wechaty-plugin-contrib 贡献 Plugin 等）
+  - Wechaty
+    - 我还计划于年末为 Wechaty 社区提供一个基于 CQHTTP 的 QQ Puppet [wechaty-puppet-cq](https://github.com/lyleshaw/wechaty-puppet-cq)
+
+## 项目成果
+
+项目仓库: <https://github.com/wechaty/python-wechaty-puppet-itchat>
```
Comment (lijiarui) [inline jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md]: You can add the pip package link here as well.

**Review by lijiarui — CHANGES_REQUESTED** [review]

File: jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md (hunk 2 above)
Comment (ly1ex) [reply jekyll/_posts/2021-10-05-ospp-final-term-wechaty-itchat-puppet.md]: Done

**Review by lijiarui — APPROVED** [review]
Good job!

**Pull request merged**